assets/src/scripts/components/map/nldiMapping.js

```diff
@@ -0,0 +1,152 @@
+        map.addLayer(pointLayer);
+    };
+
+    const onEachLineFeatureAddPopUp = function(feature, layer) {
+        const popupText = 'Data Source: NHD+' +
+            '<br>Reach ComID: ' + feature.properties.nhdplus_comid;
+        layer.bindPopup(popupText);
+    };
+
+    const addLineDataToMap = function(data, style) {
+        const lineLayer = geoJson(data, {
+            onEachFeature: onEachLineFeatureAddPopUp,
+            style: style
+            });
+        lineLayer.addTo(map);
+        const features = data.features;
+        features.forEach(function(feature) {
+            allExtents.features.push(feature);
+        });
+//        map.fitBounds(geoJson(allExtents).getBounds());
+    };
+
+    const addNldiLinesToMap = function(endpointUrl, style) {
+        get(endpointUrl)
+            .then((responseText) => {
+                addLineDataToMap(JSON.parse(responseText), style);
+            })
+            .catch(reason => {
+                console.error(reason);
+            });
+    };
+
+    const addNldiPointsToMap = function(endpointUrl, style) {
+        get(endpointUrl)
+            .then((responseText) => {
+                addPointDataToMap(JSON.parse(responseText), style);
+            })
+            .catch(reason => {
+                console.error(reason);
+            });
+    };
+
+    const upStreamSites = nldiUrl + '/'+ featureSource + '/' + featureId + '/navigate/' + upstreamNavigation + '/' + dataSource + distanceParam;
+    const downStreamSites = nldiUrl + '/'+ featureSource + '/' + featureId + '/navigate/' + downstreamNavigation + '/' + dataSource + distanceParam;
+    const upStreamFlow = nldiUrl + '/'+ featureSource + '/' + featureId + '/navigate/' + upstreamNavigation + distanceParam;
+    const downStreamFlow = nldiUrl + '/'+ featureSource + '/' + featureId + '/navigate/' + downstreamNavigation + distanceParam;
+
+    const nldiLines = [
+        {url : upStreamFlow, style : upstreamLineStyle},
+        {url : downStreamFlow, style : downstreamLineStyle}
+    ];
+
+    const nldiPoints = [
+        {url : upStreamSites, style : geojsonMarkerOptions},
+        {url : downStreamSites, style : geojsonMarkerOptions}
+    ];
+
+    nldiLines.forEach(function(pair) {
+        addNldiLinesToMap(pair.url, pair.style);
+    });
+
+    nldiPoints.forEach(function(pair) {
+        addNldiPointsToMap(pair.url, pair.style);
+    });
+
+    const legendListContainer = select(legendControl.getContainer()).select('.legend-list-container');
+    const nldiLegendList = legendListContainer.append('ul')
+                .attr('id', 'nldi-legend-list')
+                .classed('usa-list--unstyled', true);
+
+    nldiLegendList.append('li')
+                .classed('nldi-legend', true)
+                .html('<li><i style="background:' + upstreamColor + '; width: 16px; height: 16px; float: left; opacity: ' + flowLineOpacity + '; margin-right: 2px;"></i><span>Upstream Flowline</span> </li>'
```

Comment:
Consider adding each <li> element using D3.

So it would look something like this
`
nldiLegendList.append('li')
     .append('i')
           .attr('style', 'background: ....')
nldiLegendList.append('li')
     .append('i')
           .attr('class', 'fas fa-circle')
           .attr('style', 'style that you want')
`
and so one.